flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -246,7 +258,18 @@ protected StreamTask(
 	 * @param context context object for collaborative interaction between the action and the stream task.
 	 * @throws Exception on any problems in the action.
 	 */
-	protected abstract void performDefaultAction(ActionContext context) throws Exception;
+	protected void performDefaultAction(ActionContext context) throws Exception {
+		if (!inputProcessor.processInput()) {
+			if (inputProcessor.isFinished()) {
+				context.allActionsCompleted();
+			}
+			else {
+				context.actionsUnavailable();
+				inputProcessor.isAvailable().thenRun(
+					() -> ExceptionUtils.suppressExceptions(context::actionsAvailable));
```

Comment:
This probably would be addressed by @StefanRRichter's changes. But just in case, what about the following version:
```java
inputProcessor.isAvailable().thenRunAsync(
    () -> ExceptionUtils.suppressExceptions(context::actionsAvailable), workerExecutor);
context.actionsUnavailable();
```

Reply:
Yes, I've already changed this accordingly to the @StefanRRichter changes.